org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -226,79 +245,101 @@ private void runAnalysisAndUpdateMarkers(Map<IFile, IDocument> docPerFiles, fina
+    Collection<IAnalysisConfigurator> configurators = SonarLintCorePlugin.getExtensionTracker().getAnalysisConfigurators();
+    DefaultPreAnalysisContext context = new DefaultPreAnalysisContext(project, extraProperties, filesToAnalyze, tempDir);
+    for (IAnalysisConfigurator configurator : configurators) {
       if (configurator.canConfigure(project)) {
-        configurator.configure(configuratorRequest, monitor);
+        configurator.configure(context, monitor);
         usedConfigurators.add(configurator);
       }
     }
 
     return usedConfigurators;
   }
 
-  private void updateMarkers(Map<IFile, IDocument> docPerFile, Map<IResource, List<Issue>> issuesPerResource, AnalysisResults result, TriggerType triggerType)
+  private void updateMarkers(@Nullable Server server, Map<ISonarLintFile, IDocument> docPerFile, Map<ISonarLintIssuable, List<Issue>> issuesPerResource, AnalysisResults result,
+    TriggerType triggerType)
     throws CoreException {
     Set<IFile> failedFiles = result.failedAnalysisFiles().stream().map(ClientInputFile::<IFile>getClientObject).collect(Collectors.toSet());
-    Map<IResource, List<Issue>> successfulFiles = issuesPerResource.entrySet().stream()
+    Map<ISonarLintIssuable, List<Issue>> successfulFiles = issuesPerResource.entrySet().stream()
       .filter(e -> !failedFiles.contains(e.getKey()))
       // TODO handle non-file-level issues
-      .filter(e -> e.getKey() instanceof IFile)
+      .filter(e -> e.getKey() instanceof ISonarLintFile)
       .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
 
-    trackIssues(docPerFile, successfulFiles, triggerType);
+    trackIssues(server, docPerFile, successfulFiles, triggerType);
   }
 
-  private void trackIssues(Map<IFile, IDocument> docPerFile, Map<IResource, List<Issue>> rawIssuesPerResource, TriggerType triggerType) throws CoreException {
+  private void trackIssues(@Nullable Server server, Map<ISonarLintFile, IDocument> docPerFile, Map<ISonarLintIssuable, List<Issue>> rawIssuesPerResource, TriggerType triggerType)
+    throws CoreException {
 
-    String localModuleKey = getSonarProject().getProject().getName();
+    String localModuleKey = getProject().getName();
 
-    for (Map.Entry<IResource, List<Issue>> entry : rawIssuesPerResource.entrySet()) {
-      IResource resource = entry.getKey();
-      IDocument documentOrNull = docPerFile.get((IFile) resource);
+    for (Map.Entry<ISonarLintIssuable, List<Issue>> entry : rawIssuesPerResource.entrySet()) {
+      ISonarLintFile resource = (ISonarLintFile) entry.getKey();
+      IDocument documentOrNull = docPerFile.get((ISonarLintFile) resource);
       final IDocument documentNotNull;
       if (documentOrNull == null) {
-        try (TextFileContext context = new TextFileContext((IFile) resource)) {
-          documentNotNull = context.getDocument();
-        }
+        documentNotNull = ((ISonarLintFile) resource).getDocument();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "ISonarLintFile". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)